docs/architecture/adr-018-dynamic-capability-store.md

```diff
@@ -0,0 +1,197 @@
+# ADR 18: Dynamic Capability Store
+
+## Changelog
+
+- 12 December 2019: Initial version
+
+## Context
+
+Full implementation of the [IBC specification](https://github.com/cosmos/ics) requires the ability to create and authenticate object-capability keys at runtime (i.e., during transaction execution),
```

Comment:
Great job in explaining the context and use case 👏

Reply:
Thanks. I'll also add a concrete initialisation example like @fedekunze suggested.